core/main/src/Utils/ColorUtils.ts

```diff
@@ -479,16 +479,33 @@ export class ColorUtils {
         };
     }
 
-    static replaceColorSvg(image: IImage, color: IHsl, opacity: number): string {
-        if (!image.svgData) {
-            return "";
+    static replaceColorSvg(imageShape: IImage, color: IHsl, opacity: number): string {
+        if (!imageShape) {
+            throw new Error("Error tsParticles - No image provided.");
+        }
+
+        const { source, svgData } = imageShape;
+        if (!source || !svgData) {
+            throw new Error("Error tsParticles - No image.src");
+        }
+
+        const imageType = source.substr(source.length - 3);
+        if (imageType !== "svg") {
+            throw new Error("Error tsParticles - replaceColor property provided to non-.svg image.src");
         }
 
         /* set color to svg element */
-        const svgXml = image.svgData;
-        const rgbHex = /#([0-9A-F]{3,6})/gi;
+        if (svgData.includes("fill")) {
```

Comment:
I don't understand this if, why are you searching for `fill`?

Reply:
The icons @ https://simpleicons.org/ I'm using have no pre-set fill, so when the previous code ran looking for a rgbHex it found none, therefore it didn't change the color of the SVG, leaving it as the default black.

Reply:
Ah ok, if there's no fill you'll add it, am I right? This is a good idea!